src/gui/util_gui.cpp

```diff
@@ -348,18 +348,22 @@ namespace Util {
 	
 	void Marker::drawCenterMarker(QPainter* painter, const QPointF& center)
 	{
-		const auto saved_hints = painter->renderHints();
-		painter->setRenderHint(QPainter::Antialiasing, true);
+		painter->save();
```

Comment:
Note that `QPainter::save()` is convenient but will [allocate](https://code.woboq.org/qt5/qtbase/src/gui/painting/qpainter.cpp.html#_ZN8QPainter4saveEv) a new `QPainterState`. I try to avoid it when I can - which usually means when I touch just hints, pen and/or brush, as in this function.

Reply:
Before I look into addressing implicit rounding, style polishing and micro-optimizations, there is one topic which may have escaped your attention. This code is the only instance which bothers itself with assembling markers from graphic primitives. Markers displayed during line editing are scaled bitmaps provided by `PointHandles` class. Isn't extending and reusing `PointHandles` for the center markers more efficient?

Reply:
I didn't want to insist on a change here. It is just that I became careful about what I do i during `paint()`. I can't say that `PointHandles` has been designed or reviewed from this point of view. So I wouldn't recommend an immediate switch to that class at the moment. However, having these "decorations" in a single place would facilitate future improvements.

Reply:
Using the more lightweight `setRenderHint()` over the catch-all `Painter::save()` makes sense to me.